airbyte-connector-builder-server/connector_builder/impl/low_code_cdk_adapter.py

```diff
@@ -0,0 +1,43 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Dict, Iterable, List
+
+from airbyte_cdk.models import AirbyteMessage, ConfiguredAirbyteCatalog
+from airbyte_cdk.sources.declarative.declarative_stream import DeclarativeStream
+from airbyte_cdk.sources.declarative.yaml_declarative_source import ManifestDeclarativeSource
+from airbyte_cdk.sources.streams.http import HttpStream
+
+
+class LowCodeSourceAdapter:
+    def __init__(self, manifest: Dict[str, Any]):
+        # Request and response messages are only emitted for a sources that have debug turned on
+        self._source = ManifestDeclarativeSource(manifest, debug=True)
+
+    def get_http_streams(self, config: Dict[str, Any]) -> List[HttpStream]:
+        return [
+            stream.retriever
+            for stream in self._source.streams(config=config)
+            if isinstance(stream, DeclarativeStream) and isinstance(stream.retriever, HttpStream)
```

Comment:
this if statement is a little spooky. Why wouldn't it always be true? and what does it mean if it's not true? Do we lose something by removing?

Reply:
Yeah, I don't love this either. Knowing how the code works we always know that its a `DeclarativeStream`. The issue is that the `manifest_declarative_source.streams()` overrides the `abstract_source.streams()` method. And that method interface definition is just `List[Stream]`.

We can modify `manifest_declarative_source.streams()` to return a `List[DeclarativeStream]` which still adheres to the interface since declarative streams are subclasses of Streams. We will still always need to check that the retriever is an `HttpStream`. We only have one type of retriever (SimpleRetriever) which is always an HttpStream at this moment, but that could change in the future. And other retriever types might not be HttpStreams and may not make sense to have URLs.

Reply:
yup good point will do!